src/ConfigR/BasicConfig.Dictionary.cs

```diff
@@ -57,7 +59,7 @@ public bool ContainsKey(string key)
 
         public bool Remove(string key)
         {
-            log.TraceFormat(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource());
+            log.TraceFormat(string.Format(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource()));

```

Comment:
Sure, but if we discard `CultureInfo.CurrentCulture` then the call becomes:

```C#
log.TraceFormat("Removing '{0}' from {1}", key, this.GetSource());
```

There is no redundant evaluation taking place because we are passing the arguments to LibLog and it will only use them to format a string message if the log level is enabled.

Deferred execution via lambdas is only useful when you are using some other operation to generate a string, and the cost of allocating a closure is outweighed by the cost of generating the string. I.e. you would never write

```C#
logger.Debug( () => "hello world");
```

because the cost of allocating the closure will outweigh the cost of embedding the string literal "hello world" in the output assembly.